### DISCUSSION TITLE
v7, watch + TS2589: Type instantiation is excessively deep and possibly infinite.

### DISCUSSION THREAD

**Discussion by ptmkenny** [react-hook-form/react-hook-form#4647] [Q&A]
My v6 code watched a single field like this: 

`  const selectedGroup = watch('group');`

Upon upgrading to v7 and [following the migration guide](https://react-hook-form.com/migrate-v6-to-v7/), I got this error:

` TS2589: Type instantiation is excessively deep and possibly infinite.`

Based on the guide, I tried this:

`  const [selectedGroup] = watch(['group']);`

But I still get the same error.

How can I fix this?

**Comment by jorisre** [comment]
Hi @ptmkenny 

Can you provide a codesandbox with your problem please ?

**Reply by zgid123** [reply]
hi @jorisre, I faced that issue too.

It's only appear when we are using self reference interface and useFormContext.

Here a quick codesandbox: https://codesandbox.io/s/test-type-ln8lr?file=/src/Test.tsx

**Reply by bluebill1049** [reply]
that's a limitation with string template literal, it doesn't work with recursive type look up.

**Reply by zgid123** [reply]
@bluebill1049 thanks for your rely. I just realized that the watch from useForm does not raise error (ts) at first, but when we use the variable, it will raise error like useFormContext.

**Reply by bluebill1049** [reply]
can you share a CSB to reproduce the issue?

**Reply by zgid123** [reply]
@bluebill1049 forget that. I asked the solution because when I worked on the form, the `watch` from `useForm` didn't raise type error, only from `useFormContext` raised. But now it does, don't know why. This is the first time I used `react-hook-form` with self-reference interface.

Thank you so much

**Reply by bluebill1049** [reply]
np 👍
